libopenage/renderer/camera/camera.cpp

```diff
@@ -102,15 +102,37 @@ void Camera::look_at_coord(coord::scene3 coord_pos) {
 	this->look_at_scene(scene_pos);
 }
 
-void Camera::move_to(Eigen::Vector3f scene_pos) {
-	// TODO: Check and set bounds for where the camera can go and check them here
+void Camera::move_to(Eigen::Vector3f scene_pos,
+                     std::optional<std::pair<float, float>> x_bounds,
+                     std::optional<std::pair<float, float>> z_bounds) {
+	// Calculate look at coordinates using same method as in look_at_scene(..)
+	auto y_delta = this->scene_pos[1] - scene_pos[1];
+	auto xz_distance = y_delta * std::numbers::sqrt3;
+
+	float side_length = xz_distance / std::numbers::sqrt2;
+	scene_pos[0] += side_length;
+	scene_pos[2] += side_length;
```

Comment:
`move_to` should not do the look at calculations because it is intended to just move the camera to a position in the scene. Looking at a scene coordinate is already implemented in `look_at_scene`. This might break existing code where `move_to` is currently used.

The changes here might also break `look_at_scene`'s behavior, since the method calls `move_to` but with the already computed camera position. Therefore, the look at calculations would be done a second time on the already translated camera position.